frontend/src/Dashboard/CrossingDetailPage/CrossingDetailPage.js

```diff
@@ -1,19 +1,76 @@
 import React, { Component } from 'react';
+import { graphql, compose } from 'react-apollo';
+import gql from 'graphql-tag';
 import CrossingDetails from './CrossingDetails';
 import CrossingStatusHistory from './CrossingStatusHistory';
 
+
 class CrossingDetailPage extends Component {
   render() {
-    const crossingId = this.props.match.params.id;
+    if ( !this.props.CrossingByIdQuery ||
+          this.props.CrossingByIdQuery.loading ||
+          !this.props.CrossingHistoryQuery ||
+          this.props.CrossingHistoryQuery.loading ) {
+      return (<div>Loading</div>)
+    }
+
+    const crossing = this.props.CrossingByIdQuery.crossingById;
+    const community = crossing.communityCrossingsByCrossingId.nodes[0].communityByCommunityId;
+    const history = this.props.CrossingHistoryQuery.allStatusUpdates.nodes;
 
     return (
       <div>
-        <CrossingDetails crossingId={crossingId}/>
-        <CrossingStatusHistory crossingId={crossingId}/>
+        <CrossingDetails crossing={crossing} community={community}/>
+        <CrossingStatusHistory history={history}/>
       </div>
     );
   }
 
 }
 
-export default CrossingDetailPage;
+const CrossingByIdQuery = gql`
+  query crossingById($crossingId:Int!) {
+    crossingById(id:$crossingId) {
+      name
+      humanCoordinates
+      humanAddress
+      description
+      communityCrossingsByCrossingId {
+        nodes {
+          communityByCommunityId {
+            name
+          }
+        }
+      }
+    }
+  }
+`;
+
+const CrossingHistoryQuery = gql`
+  query crossingHistory($crossingId:Int!) {
+    allStatusUpdates(condition:{crossingId:$crossingId}) {
+      nodes {
+        id
+      }
+    }
+  }
+`;
+
+export default compose(
+  graphql(CrossingByIdQuery, {
+    name: 'CrossingByIdQuery',
+    options: (ownProps) => ({
+      variables: {
+        crossingId: ownProps.match.params.id
+      }
+    })
+  }),
+  graphql(CrossingHistoryQuery, {
```

Comment:
Cool to see us using `compose` here to combine two queries. Wondering when we should be thinking ahead about pagination on the crossing history. Seems like it could be a bottleneck.

Reply:
I'm pretty sure we'll be able to handle pagination through variables - if we keep the page variable in state on CrossingDetailPage we can pass it down as a prop and interact with it in any subcomponents.